var/spack/repos/builtin/packages/lbann/package.py

```diff
@@ -410,6 +423,12 @@ def cmake_args(self):
                         arch_str, cxxflags_str)
                 )
 
+        # IF IBM ESSL is used it needs help finding the proper LAPACK libraries
+        if self.spec['essl'] in spec:
+            args.extend([
+                '-DLAPACK_LIBRARIES=%s;-llapack;-lblas' % self.spec['essl'].libs,
+                '-DBLAS_LIBRARIES=%s;-lblas' % self.spec['essl'].libs])
```

Comment:
That is a good point, but at the moment, ESSL really only exports one library.  I also think that this is really more of a bug in the ESSL package or in CMake's find BLAS/LAPACK functions.

Reply:
The API of `LibraryList` will insert spaces. I think if you do `% ';'.join(self.spec[essl].libs.names)' it would future proof it to any libraries being added.

Reply:
Or wait that by itself won't work since `.names` doesn't convert to `-lfoo` just `foo`. So

```
';'.join('-l{0}'.format(lib) for lib in self.spec[essl].libs.names)
```

Reply:
Thanks @scheibelp I have applied these changes.